Update `README.md`. (#39)
w3c-ccg/di-ed25519signature2020-test-suite: README.md

# di-ed25519signature2020-test-suite

> > Test Suite for [Ed25519Signature2020](https://w3c-ccg.github.io/lds-ed25519-2020/).

Author: This URL is giving a 404. Should it be pointing to `https://github.com/w3c/vc-di-eddsa` instead?

Reviewer: Yes, specifically, use this URL: https://www.w3.org/TR/vc-di-eddsa/#the-ed25519signature2020-suite

> # di-ed25519signature2020-test-suite
> 
> > Test Suite for [Ed25519Signature2020](https://w3c-ccg.github.io/lds-ed25519-2020/).

Reviewer 2: URL may still need to be changed.
```suggestion
# Test Suite for [Ed25519Signature2020](https://w3c-ccg.github.io/lds-ed25519-2020/)
```

Author: @msporny Thanks, fixed the URL [71777d5](https://github.com/w3c-ccg/di-ed25519signature2020-test-suite/pull/39/commits/71777d5dea603ded1cba9f021bfd20296fbfa19d)

## Table of Contents

- [Background](#background)
- [Install](#install)
- [Usage](#usage)
- [Implementation](#implementation)
- [Docker Integration](#docker-integration-todo)
- [License](#license)

## Background
Provides interoperability tests for verifiable credential processors

> (issuers/verifiers) that support [Ed25519Signature2020](https://w3c-ccg.github.io/lds-ed25519-2020/).

Reviewer 2: ```suggestion
(issuers and verifiers) that support [Ed25519Signature2020](https://w3c-ccg.github.io/lds-ed25519-2020/).
```

## Usage

```
npm test
```

## Implementation

You will need an issuer and verifier that are compatible with [VC API](https://w3c-ccg.github.io/vc-api/)
and are capable of handling issuance and verification of Verifiable Credentials
with `Ed25519Signature2020` proof type.

To add your implementation to this test suite, you will need to add 2 endpoints
to your implementation manifest.

> - A credentials issuer endpoint (`/credentials/issue`) in the `issuers`
>   property.
> - A credentials verifier endpoint (`/credentials/verify`) in the `verifiers`

Reviewer 2: ```suggestion
- A credential issuer endpoint (`/credentials/issue`) in the `issuers`
  property.
- A credential verifier endpoint (`/credentials/verify`) in the `verifiers`
```

  property.

All endpoints will need the tag `Ed25519Signature2020`.

A simplified manifest would look like this:

```js
{
  "name": "My Company",
  "implementation": "My implementation",
  "issuers": [{
    "id": "",
    "endpoint": "https://issuer.mycompany.com/credentials/issue",
    "method": "POST",
    "tags": ["Ed25519Signature2020"]
  }],
  "verifiers": [{
    "id": "",
    "endpoint": "https://verifier.mycompany.com/credentials/verify",
    "method": "POST",
    "tags": ["Ed25519Signature2020"]
  }]
}
```

> The example above represents an unauthenticated endpoint. You may add zcap or
> oauth authentication to your endpoints. You can find an example in the
> [vc-api-test-suite-implementations README here](https://github.com/w3c-ccg/vc-api-test-suite-implementations#adding-a-new-implementation).

Reviewer 2: ```suggestion
The example above represents an unauthenticated endpoint. You may add ZCAP or
OAuth authentication to your endpoints. You can find an example in the
[vc-api-test-suite-implementations README](https://github.com/w3c-ccg/vc-api-test-suite-implementations#adding-a-new-implementation).
```

To run the tests, some implementations may require client secrets that can be

> passed as env variables to the test script. To see which ones require client

Reviewer 2: ```suggestion
passed as environment variables to the test script. To see which implementations require client
```

secrets, please check the implementation manifest within the
[vc-api-test-suite-implementations](https://github.com/w3c-ccg/vc-api-test-suite-implementations/tree/main/implementations) library.

## Docker Integration (TODO)

We are presently working on implementing a new feature that will enable the

> utilization of Docker images instead of live endpoints. The docker image that
> you provide will be started when the test suite is run. The image is expected
> to expose the API provided above, which will be utilized in the same way that

Reviewer 2: ```suggestion
use of Docker images instead of live endpoints. The Docker image that
you provide will be started when the test suite is run. The image is expected
to expose the API provided above, which will be used in the same way that
```
